packages/studio-base/src/players/FoxgloveWebSocketPlayer/index.ts

```diff
@@ -160,6 +161,11 @@ export default class FoxgloveWebSocketPlayer implements Player {
     }
     log.info(`Opening connection to ${this._url}`);
 
+    // Set a timeout to abort the connection if we are still not connected by then
```

Comment:
Worth adding the _why_ here. Something along the lines of us noticing that sometimes misbehaving network setups hang the connection attempt as if still attempting but will never actually complete the attempt. So instead we kill our attempt and try the entire connection again.

Reply:
```suggestion
    // Set a timeout to abort the connection if we are still not connected by then.
    // This will abort hanging connection attempts that can for whatever reason not
    // establish a server with the server.
```

Open to other suggestions